### ISSUE TITLE
Does i18next follow semver?

### ISSUE THREAD

**Issue by traverse** [i18next/i18next#1342]
Hi, I ran into an issue today when trying to upgrade `i18next@17.0.6` -> `i18next@17.2.0`. I expected this version to be backwards compatible since it is only a minor version bump. However I ran into some issues where the TypeScript compiler required `esModuleInterop` to be turned on which I can't do easily because it would require me to fix for example every Lodash import.

The issue is Lodash exports it functions like this `module.exports = get;` which requires you to import it as follows `import * as get from 'lodash/get';` without `esModuleInterop`. `i18next` has an esm export which is `export default i18next;` and that is the one being used. So my options are: `// @ts-ignore` every `i18next` import, turn on `esModuleInterop` and fix every other import that doesn't work because of it or another suggestion I saw in similar issues was to use `import * as i18next from 'i18next'` which won't work since it is not same as importing the default export the application just breaks.

This leads me to the conclusion that either semver isn't being followed at all or I'm doing something completely wrong in which case I could require some more help on how to fix this.

**Comment by rosskevin** [comment]
Yes i18next follows semver for the runtime.  No it does not strictly follow this for typescript, but we try to line those releases up where possible.

The `esModuleInterop` work was not a breaking change unless it was being used incorrectly.

We test both `esModuleInterop` and non-`esModuleInterop`.  For examples of non-`esModuleInterop` see https://github.com/i18next/i18next/tree/master/test/typescript/nonEsModuleInterop

If you have a case that breaks, please PR a breaking test to this directory, tag me and I will look at it.  You can also look at our tsconfig for guidance: https://github.com/i18next/i18next/blob/master/tsconfig.nonEsModuleInterop.json
